Jenkins_setup.md

```diff
@@ -0,0 +1,161 @@
+then the CI may quickly run out of GitHub access bandwidth, which will stall the CI.
+
+For Kata Containers CI we configure Jenkins with a GitHub [Personal API token](https://blog.github.com/2013-05-16-personal-api-tokens/), which increases the bandwidth capacity.
+
+## Job build script
+
+Kata Containers Jenkins builds, for both PR and Master builds, are invoked using the top
+level [`jenkins_job_build.sh`](https://github.com/kata-containers/tests/blob/master/.ci/jenkins_job_build.sh) script from the [katacontainers/tests](https://github.com/kata-containers/tests)
+repository.
+
+This script takes a single argument (the path to the repository to build), and utilises
+some environment variables to guide what sort of build and run it should perform.
+
+An example of how this script is invoked from Jenkins is shown in the [Setting up Jenkins](#setting-up-jenkins) section.
+
+## Setting up Jenkins
+
+### Naming jobs
+
+Generally the Jobs in Jenkins are named in the form `kata-containers-<repo>-<distro>-<buildtype>`,
+e.g., for a runtime repo PR build on Ubuntu 17.10, the Job would be called:
+`kata-containers-runtime-ubuntu-17-10-PR`. Following this naming convention will make it much
+easier for other members of the project to navigate all CIs in a similar manner.
+
+### Setting up the scripts
+
+Both PR and Master builds use the same form of build and post-build teardown scripts that
+are configured in the Jenkins Job UI. These scripts look like:
+
+Build execute shell:
+```bash
+#!/bin/bash
+
+set -e
+
+export ghprbPullId
+export ghprbTargetBranch
+
+cd $HOME
+git clone https://github.com/kata-containers/tests.git
+cd tests
+
+.ci/jenkins_job_build.sh "github.com/kata-containers/runtime"
+```
+> **Note:**
+>
+> The `export` of the `ghprb` variables for the Master builds is benign, as the Master
+> builds do not use the GHPRB plugin. Subsequently, the `jenkins_job_build.sh` script can
+> still test for the GHPRB varibles to distinguish between a PR and Master build.
+
+This script is placed into the Jenkins build dialog like:
+
+![Jenkins Build dialog](pictures/jenkins_build.png)
+
+Teardown script:
+
+```bash
+#!/bin/bash
+
+export GOPATH=$WORKSPACE/go
+
+cd $GOPATH/src/github.com/kata-containers/tests
+.ci/teardown.sh "$WORKSPACE"
+```
+
+The teardown script is entered into the Jenkins Post-Build Actions dialog like:
+
+![Jenkins Build dialog](pictures/jenkins_build.png)
+
+## Environment variables
+
+The `jenkins_job_build.sh` script both checks for some variables to decide what to do,
+and sets some variables for sub-scripts to check.
+
+### Input variables
+
+The following variables influence the actions of the script:
+
+| Var                 | Effect                                                           |
+| ---------           | ---------------------------------------------------------------- |
+| `WORKSPACE`         | Must be set. Is set by Jenkins to point at the directory the script will use as its workspace (scratch area) |
```

Comment:
Suggested rewrite to stay active: Is set by Jenkins to point at the directory the script uses as its workspace (scratch